docs/src/examples/ChartManager.ts

```diff
@@ -0,0 +1,268 @@
+    } else {
+      this._listeners.set(elementId, new Map([[0, listener]]));
+    }
+
+    return listenerId;
+  }
+
+  public removeListener(elementId: number, listenerId: number): void {
+    this._listeners.get(elementId).delete(listenerId);
+  }
+
+  public clearListeners(): void {
+    this._listeners.clear();
+  }
+
+  public addElement(
+    label: State | string = null,
+    subtext: string | null = null,
+    isVisible: boolean = true,
+    isHeader: boolean = false
+  ): [(isActive: boolean) => void, number] {
+    const newId = this._elements.length;
+
+    if (typeof label == 'number') {
+      label = stateToName.get(label);
+    }
+
+    const newElementData = {
+      id: newId,
+      label: label,
+      subtext: subtext,
+      position: null,
+      isVisible: isVisible,
+      isHeader: isHeader,
+    };
+
+    this._elements.push(newElementData);
+
+    // this callback will be used by a .onX hook to broadcast this event to all listeners
+    return [
+      (isActive: boolean) => {
+        this._listeners.get(newId)?.forEach((listener) => listener(isActive));
+      },
+      newId,
+    ];
+  }
+
+  public addHeader(text: string): number {
+    // todo: add elements which can display text, and fill in all of the assigned space
+    return this.addElement(text, null, true, true)[1];
+  }
+
+  public addConnection(fromId: number, toId: number) {
+    this._connections.push({
+      id: this._connections.length,
+      from: fromId,
+      to: toId,
+    });
+  }
+
+  public connectAll(handle: GestureHandle) {
+    stateConnectionsMap.forEach(([fromState, toState]) => {
+      const fromId = handle.elementIds.get(fromState);
+      const toId = handle.elementIds.get(toState);
+      if (fromId && toId) {
+        this.addConnection(fromId, toId);
+      }
+    });
+  }
+
+  public newGesture(
+    gesture: GesturesUnion
+  ): [GestureHandle, GesturesUnion, any] {
+    const [beganCallback, beganId] = this.addElement(State.BEGAN);
+    const [activeCallback, activeId] = this.addElement(State.ACTIVE);
+    const [endCallback, endId] = this.addElement(State.END);
+    const [failedCallback, failedId] = this.addElement(State.FAILED);
+    const [cancelledCallback, cancelledId] = this.addElement(State.CANCELLED);
+    const [undeterminedCallback, undeterminedId] = this.addElement(
+      State.UNDETERMINED
+    );
+
+    const handle = new GestureHandle();
+
+    handle.elementIds.set(State.BEGAN, beganId);
+    handle.elementIds.set(State.ACTIVE, activeId);
+    handle.elementIds.set(State.END, endId);
+    handle.elementIds.set(State.FAILED, failedId);
+    handle.elementIds.set(State.CANCELLED, cancelledId);
+    handle.elementIds.set(State.UNDETERMINED, undeterminedId);
+
+    undeterminedCallback(true);
+
+    const WAVE_DELAY_MS = 100;
+
+    const resetAllStates = (event: GestureStateChangeEvent<any>) => {
+      undeterminedCallback(true);
+      if (event.state == 1) {
```

Comment:
```suggestion
      if (event.state == State.FAILED) {
```
We should use `State` enum.